crates/goose-server/src/routes/agent.rs

```diff
@@ -255,7 +255,7 @@ async fn update_agent_provider(
             .get_param("GOOSE_MODEL")
             .expect("Did not find a model on payload or in env to update provider with")
     });
-    let model_config = ModelConfig::new(model);
+    let model_config = ModelConfig::new(&model).map_err(|_| StatusCode::INTERNAL_SERVER_ERROR)?;
```

Comment:
the line above where we have .expect() will kill goosed and it is all over. not sure how we would have gotten here, but still, we should probably do another 500; I also think we should change the description of the 500 from internal server error here to something descriptive; could not update provider, make sure that you have the right config or something